// auth0/dist/cjs/management/__generated/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagementClientBase = void 0;
__exportStar(require("./managers/index.js"), exports);
__exportStar(require("./models/index.js"), exports);
const index_js_1 = require("./managers/index.js");
class ManagementClientBase {
    constructor(configuration) {
        this.configuration = configuration;
        this.actions = new index_js_1.ActionsManager(this.configuration);
        this.anomaly = new index_js_1.AnomalyManager(this.configuration);
        this.attackProtection = new index_js_1.AttackProtectionManager(this.configuration);
        this.blacklists = new index_js_1.BlacklistsManager(this.configuration);
        this.branding = new index_js_1.BrandingManager(this.configuration);
        this.clientGrants = new index_js_1.ClientGrantsManager(this.configuration);
        this.clients = new index_js_1.ClientsManager(this.configuration);
        this.connections = new index_js_1.ConnectionsManager(this.configuration);
        this.customDomains = new index_js_1.CustomDomainsManager(this.configuration);
        this.deviceCredentials = new index_js_1.DeviceCredentialsManager(this.configuration);
        this.emailTemplates = new index_js_1.EmailTemplatesManager(this.configuration);
        this.emails = new index_js_1.EmailsManager(this.configuration);
        this.grants = new index_js_1.GrantsManager(this.configuration);
        this.guardian = new index_js_1.GuardianManager(this.configuration);
        this.hooks = new index_js_1.HooksManager(this.configuration);
        this.jobs = new index_js_1.JobsManager(this.configuration);
        this.keys = new index_js_1.KeysManager(this.configuration);
        this.logStreams = new index_js_1.LogStreamsManager(this.configuration);
        this.logs = new index_js_1.LogsManager(this.configuration);
        this.organizations = new index_js_1.OrganizationsManager(this.configuration);
        this.prompts = new index_js_1.PromptsManager(this.configuration);
        this.resourceServers = new index_js_1.ResourceServersManager(this.configuration);
        this.roles = new index_js_1.RolesManager(this.configuration);
        this.rules = new index_js_1.RulesManager(this.configuration);
        this.rulesConfigs = new index_js_1.RulesConfigsManager(this.configuration);
        this.stats = new index_js_1.StatsManager(this.configuration);
        this.tenants = new index_js_1.TenantsManager(this.configuration);
        this.tickets = new index_js_1.TicketsManager(this.configuration);
        this.userBlocks = new index_js_1.UserBlocksManager(this.configuration);
        this.users = new index_js_1.UsersManager(this.configuration);
        this.usersByEmail = new index_js_1.UsersByEmailManager(this.configuration);
    }
}
exports.ManagementClientBase = ManagementClientBase;
