// auth0/dist/cjs/auth/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticationClient = exports.AuthApiError = exports.IdTokenValidatorError = void 0;
const database_js_1 = require("./database.js");
const oauth_js_1 = require("./oauth.js");
const passwordless_js_1 = require("./passwordless.js");
__exportStar(require("./database.js"), exports);
__exportStar(require("./oauth.js"), exports);
__exportStar(require("./passwordless.js"), exports);
var id_token_validator_js_1 = require("./id-token-validator.js");
Object.defineProperty(exports, "IdTokenValidatorError", { enumerable: true, get: function () { return id_token_validator_js_1.IdTokenValidatorError; } });
var base_auth_api_js_1 = require("./base-auth-api.js");
Object.defineProperty(exports, "AuthApiError", { enumerable: true, get: function () { return base_auth_api_js_1.AuthApiError; } });
class AuthenticationClient {
    constructor(options) {
        this.database = new database_js_1.Database(options);
        this.oauth = new oauth_js_1.OAuth(options);
        this.passwordless = new passwordless_js_1.Passwordless(options);
    }
}
exports.AuthenticationClient = AuthenticationClient;
